src/main/groovy/au/ala/org/ws/security/AlaWsSecurityGrailsPluginConfiguration.groovy

```diff
@@ -1,45 +1,40 @@
 package au.ala.org.ws.security
 
-import au.org.ala.ws.security.JwtAuthenticator
+
+
 import au.org.ala.ws.security.JwtProperties
 import au.org.ala.ws.security.Pac4jProfileManagerHttpRequestWrapperFilter
-import com.nimbusds.jose.jwk.source.JWKSource
-import com.nimbusds.jose.jwk.source.RemoteJWKSet
-import com.nimbusds.jose.proc.SecurityContext
-import com.nimbusds.jose.util.DefaultResourceRetriever
-import com.nimbusds.jose.util.ResourceRetriever
-import com.nimbusds.openid.connect.sdk.op.OIDCProviderMetadata
-import grails.util.Metadata
-import org.pac4j.core.authorization.generator.FromAttributesAuthorizationGenerator
+
 import org.pac4j.core.client.Client
 import org.pac4j.core.config.Config
 import org.pac4j.core.context.WebContextFactory
 import org.pac4j.core.context.session.SessionStore
-import org.pac4j.core.engine.DefaultSecurityLogic
-import org.pac4j.http.client.direct.DirectBearerAuthClient
 import org.pac4j.jee.context.JEEContextFactory
 import org.pac4j.jee.context.session.JEESessionStore
-import org.pac4j.jee.filter.SecurityFilter
+import org.pac4j.oidc.config.OidcConfiguration
 import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Value
 import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean
 import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
 import org.springframework.boot.context.properties.EnableConfigurationProperties
 import org.springframework.boot.web.servlet.FilterRegistrationBean
 import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.ComponentScan
 import org.springframework.context.annotation.Configuration
-import org.springframework.core.Ordered
-
-import javax.servlet.DispatcherType
 
 @Configuration
 @EnableConfigurationProperties(JwtProperties)
+@ComponentScan('au.org.ala.ws.security')
```

Comment:
I don't particularly like this, component scan is slow when your contains a lot of JARS, I'd rather keep all the Spring stuff in this `@Configuration` class, remove Spring annotations from the actual classes and then use `spring.factories` pointing at this class.  This way you don't have to inject Properties classes, etc, the classes themselves remain properly testable without spring, etc, etc.